src/TurboMqtt/Client/ClientStreamOwner.cs

```diff
@@ -466,6 +466,9 @@ private async Task ReplaceTransport()
 
         // swap transports
         _client!.SwapTransport(_currentTransport);
+        
+        // Reset the ack actor connection state
+        _clientAckActor!.Tell(ClientAcksActor.Reconnect.Instance);
```

Comment:
The actual fix

Reply:
This also needs to happen if we try to do a reconnect after a first-chance connection failure, so there should be another call just like this one any time you see a `SwapTransport` call.